pkg/core/blockchain.go

```diff
@@ -1655,9 +1655,8 @@ var (
 	ErrInvalidVerificationContract = errors.New("verification contract is missing `verify` method")
 )
 
-// initVerificationVM initializes VM for witness check.
-func (bc *Blockchain) initVerificationVM(ic *interop.Context, hash util.Uint160, witness *transaction.Witness) error {
-	v := ic.VM
+// InitVerificationVM initializes VM for witness check.
+func (bc *Blockchain) InitVerificationVM(v *vm.VM, getContract func(util.Uint160) (*state.Contract, error), hash util.Uint160, witness *transaction.Witness) error {
```

Comment:
How about making `GetTestVM` return interop context along with VM itself and not changing interface here?

Reply:
Are you sure that this is what we need? Because there's an import cycle problem: Blockchainer (it includes GetTestVM) -> Context (it contains chain, but we need it to be Blockchainer) -> Blockchainer. I was trying to avoid it, but I'm not sure it's possible in this case.

Reply:
OK, that's unexpected. I'll check if we can do anything better, I don't like the interface we have here now with callbacks, it was rather simple and now it's not.

Reply:
Some real improvement would require substantial refactoring, not now.